glamox/src/Components/BooksList.jsx

```diff
@@ -1,70 +0,0 @@
-import React, { useEffect, useState } from "react";
-import { useDispatch, useSelector } from "react-redux";
-import { getBooks } from "../ReduxStore/AppStore/action";
-import { BookCard } from "./BookCard";
-import styles from "../Styles/Booklist.module.css"
-import styled from "@emotion/styled";
-
-// import styled from "styled-components";
-import { useLocation, useSearchParams } from "react-router-dom";
-import { Skeleton } from "@chakra-ui/react";
-
-export const BooksList = () => {
-  const {books,isLoading} = useSelector((store) => store.AppReducer);
-  const dispatch = useDispatch();
-  const location = useLocation();
-  const [searchParams] = useSearchParams();
-  const [cat,setCat] = useState('makeup')
-
-  //console.log(isLoading)
-  // useEffect(() => {
-  //   const order = searchParams.get("order");
-  //   let paramObj = {
-  //     params: {
-  //       brand: searchParams.getAll("brand"),
-  //       _sort: order ,
-        
-  //       _order: order, //acs or desc
-  //     },
-  //   };
-  //   console.log(order);
-  //   dispatch(getBooks(paramObj, cat));
-  // }, [location.search]);
-
-  useEffect(()=>{
-    if(location || books.length===0){
-        const sortBy = searchParams.get('sort');
-        const getBooksParams = {
-            params: {
-                brand: searchParams.getAll('brand'),
-                _sort: sortBy && 'price',
-                _order: sortBy,
-            },
-        };
-        dispatch(getBooks(getBooksParams, cat))
-    }
-},[location.search]);
-  
-  return (
-    <div  className={styles.booklist_div}>
-      {books.length > 0 &&
-        books.map((el) =>
-          isLoading? <Skeleton height='70vh' />:
-          <BookCard key={el.id} book={el} />
-        )}
```

Comment:
Avoid writing javascript logic inside jsx/tsx. you can make a function outside and then call that function here.
For example: 
`function renderBookCards() {
   {books.length > 0 &&
        books.map((el) =>
          isLoading? <Skeleton height='70vh' />:
          <BookCard key={el.id} book={el} />
        )}
}

// inside jsx
<div>
{renderBookCards()}
</div>
`